libc/main.c

```diff
@@ -15,12 +15,19 @@
 
 #include <stdlib.h>
 #include "unity_fixture.h"
+#include "common.h"
 
 /* no need for forward declarations, RUN_TEST_GROUP does it by itself */
 void runner(void)
 {
 	setenv("POSIXLY_CORRECT", "y", 1);
 
+	/* the following files may not be present on dummyfd targets,
+	create them to make libc tests common */
+	create_dir_if_missing("/tmp");
+	create_dir_if_missing("/etc");
+	create_file_if_missing("/etc/passwd", "root:0B1ANiYi45IhxkfmUW155/GBd4IRE=:0:0:root:/:/bin/sh");
```

Comment:
Did you make sure that we can use `TEST_ASEERT*` here? It is allowed to call it in runner function or maybe only in `RUN_GROUP_TEST*`?

Reply:
Yeah, it wasn't allowed. Thanks for this suggestion and please check the new approach